ugs-platform/ugs-platform-plugin-jog/src/main/java/com/willwinder/ugs/nbp/jog/actions/SetMachineDecimalFormatFour.java

```diff
@@ -0,0 +1,104 @@
+import com.willwinder.universalgcodesender.model.BackendAPI;
+import com.willwinder.universalgcodesender.model.UGSEvent;
+import com.willwinder.universalgcodesender.model.events.ControllerStateEvent;
+import com.willwinder.universalgcodesender.model.events.SettingChangedEvent;
+import java.awt.EventQueue;
+import java.awt.event.ActionEvent;
+import javax.swing.AbstractAction;
+import static javax.swing.Action.NAME;
+import javax.swing.JCheckBoxMenuItem;
+import javax.swing.JMenuItem;
+import org.openide.awt.ActionID;
+import org.openide.awt.ActionReference;
+import org.openide.awt.ActionReferences;
+import org.openide.awt.ActionRegistration;
+import org.openide.util.actions.Presenter;
+import org.openide.util.lookup.ServiceProvider;
+
+///**
+// *
+// * @author Damian Nikodem
+// */
+
+/**
+ * An action to set the number of Decimal Places in the backend.
+ *
+ * @author Damian Nikodem
+ */
+@ActionID(
+        category = LocalizingService.CATEGORY_MACHINE,
+        id = "com.willwinder.ugs.nbp.jog.actions.SetMachineDecimalFormatFour")
+@ActionRegistration(
+        displayName = "resources.MessagesBundle#platform.plugin.jog.setMachineDecimalFormatFour",
+        lazy = false)
+@ActionReferences({
+        @ActionReference(
+                path = LocalizingService.MENU_MACHINE_JOG_DECIMAL_PLACES_4,
+                position = 10000,
+                separatorAfter = 10001)
+})
+@ServiceProvider(service = SetMachineDecimalFormatFour.class)
+public class SetMachineDecimalFormatFour extends AbstractAction implements Presenter.Menu {
+    private final String DECIMAL_FORMAT_FOUR = "0.0000";
+    private final BackendAPI backend;
+    private final JCheckBoxMenuItem menuItem;
+
+    public SetMachineDecimalFormatFour() {
+        String title = Localization.getString("platform.plugin.jog.setMachineDecimalFormatFour");
+        putValue(NAME, title);
+
+        menuItem = new JCheckBoxMenuItem(title);
+        menuItem.setAction(this);
+
+        backend = CentralLookup.getDefault().lookup(BackendAPI.class);
+        backend.addUGSEventListener(this::onBackendEvent);
+        setEnabled(isEnabled());
+        onBackendEvent( new SettingChangedEvent()) ;
+    }
+
+    private void onBackendEvent(UGSEvent event) {
+        if (event instanceof SettingChangedEvent) {
+            menuItem.setSelected(backend.getSettings().getMachineDecimalFormat().equals(DECIMAL_FORMAT_FOUR));
+        } else if (event instanceof ControllerStateEvent) {
+            EventQueue.invokeLater(() -> setEnabled(isEnabled()));
+        }
+    }
+
+    @Override
+    public boolean isEnabled() {
+        return backend != null && backend.isConnected();
+    }
+
+    @Override
+    public JMenuItem getMenuPresenter() {
+        return menuItem;
+    }
+
+    @Override
+    public void actionPerformed(ActionEvent e) {
+        backend.getSettings().setMachineDecimalFormat(DECIMAL_FORMAT_FOUR) ;
+        backend.getSettings().changed();                    
```

Comment:
Move this call to changed() into the Settings::setMachineDecimalFormat, then you can hide the changed method again to private.